src/main/java/com/cst438/controller/StudentController.java

```diff
@@ -7,75 +8,151 @@
+        // user must be a student
+        // hint: use enrollment repository method findEnrollmentByStudentIdOrderByTermId
+
+        validateStudent(studentId);
+
+        List<EnrollmentDTO> enrollmentDTOs = new ArrayList<>();
+        for (Enrollment enrollment : enrollmentRepository.findEnrollmentsByStudentIdOrderByTermId(studentId)) {
+            enrollmentDTOs.add(EnrollmentDTO.fromEntity(enrollment));
+        }
+
+        return enrollmentDTOs;
+    }
 
+    // student gets a list of their enrollments for the given year, semester
+    // user must be student
+    // studentId will be temporary until Login security is implemented
+    @GetMapping("/enrollments")
+    public List<EnrollmentDTO> getSchedule(
+       @RequestParam("year") int year,
+       @RequestParam("semester") String semester,
+       @RequestParam("studentId") int studentId)
+    {
+        // hint: use enrollment repository method findByYearAndSemesterOrderByCourseId
+
+        validateStudent(studentId);
+
+        List<EnrollmentDTO> enrollmentDTOs = new ArrayList<>();
+        for (Enrollment enrollment : enrollmentRepository
+            .findByYearAndSemesterOrderByCourseId(year, semester, studentId))
+        {
+            enrollmentDTOs.add(EnrollmentDTO.fromEntity(enrollment));
+        }
+
+        return enrollmentDTOs;
+    }
 
     // student adds enrollment into a section
     // user must be student
     // return EnrollmentDTO with enrollmentId generated by database
     @PostMapping("/enrollments/sections/{sectionNo}")
     public EnrollmentDTO addCourse(
-		    @PathVariable int sectionNo,
-            @RequestParam("studentId") int studentId ) {
-
-        // TODO
-
+        @PathVariable int sectionNo,
+        @RequestParam("studentId") int studentId)
+    {
         // check that the Section entity with primary key sectionNo exists
         // check that today is between addDate and addDeadline for the section
         // check that student is not already enrolled into this section
         // create a new enrollment entity and save.  The enrollment grade will
         // be NULL until instructor enters final grades for the course.
 
-        // remove the following line when done.
-        return null;
+        Section section = sectionRepository.findById(sectionNo).orElse(null);
+
+        if (section == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Unknown sectionNo");
+        }
+
+        User user = validateStudent(studentId);
+
+        Term term = section.getTerm();
+        Date now = Date.from(Instant.now());
+
+        if (now.before(term.getAddDate()) || now.after(term.getAddDeadline())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Course cannot be added at this time");
+        }
+
+        Enrollment existingEnrollment = enrollmentRepository
+            .findEnrollmentBySectionNoAndStudentId(sectionNo, studentId);
+
+        if (existingEnrollment != null) {
```

Comment:
Alternative: `if (enrollmentRepository.findEnrollmentBySectionNoAndStudentId(sectionNo, studentId) != null) { throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Student is already enrolled in section"); }`